src/Bridges/quadtosocbridge.jl

```diff
@@ -0,0 +1,179 @@
+```math
+\\frac{1}{2}x^T Q x + a^T x + b \\le 0
+```
+is a convex set if `Q` is positive semidefinite and is the union of two convex
+cones if `a` and `b` are zero (i.e. *homogeneous* case) and `Q` has only one
+negative eigenvalue.
+
+## Non-homogeneous case
+
+If `Q` is positive semidefinite, there exists `U` such that ``Q = U^T U``, the
+inequality can then be rewritten as
+```math
+\\|U x\\|_2 \\le 2 (a^T x + b)
+```
+which is equivalent to the membership of `(1, a^T x + b, Ux)` to the rotated
+second-order cone.
+
+## Homogeneous case
+
+If `Q` has only one negative eigenvalue, the set of `x` such that ``x^T Q x \\le
+0`` is the union of a convex cone and its opposite. We can choose which one to
+model by checking the existence of bounds on variables as shown below.
+
+### Second-order cone
+
+If `Q` is diagonal and has eigenvalues `(1, 1, -1)`, the inequality
+``x^2 + x^2 \\le z^2`` combined with ``z \\ge 0`` defines the Lorenz cone (i.e.
+the second-order cone) but when combined with ``z \\le 0``, it gives the
+opposite of the second order cone. Therefore, we need to check if the variable
+`z` has a lower bound 0 or an upper bound 0 in order to determine which cone is
+
+### Rotated second-order cone
+
+The matrix `Q` corresponding to the inequality ``x^2 \\le 2yz`` has one
+eigenvalue 1 with eigenvectors `(1, 0, 0)` and `(0, 1, -1)` and one eigenvalue
+`-1` corresponding to the eigenvector `(0, 1, 1)`. Hence if we intersect this
+union of two convex cone with the halfspace ``x + y \\ge 0``, we get the rotated
+second-order cone and if we intersect it with the halfspace ``x + y \\le 0`` we
+get the opposite of the rotated second-order cone. Note that `y` and `z` have
+the same sign since `yz` is nonnegative hence ``x + y \\ge 0`` is equivalent to
+``x \\ge 0`` and ``y \\ge 0``.
+
+### Note
+
+The check for existence of bound can be implemented (but inefficiently) with the
+current interface but if bound is removed or transformed (e.g. `≤ 0` transformed
+into `≥ 0`) then the bridge is no longer valid. For this reason the homogeneous
+version of the bridge is not implemented yet.
+"""
+struct QuadtoSOCBridge{T} <: AbstractBridge
+    soc::CI{MOI.VectorAffineFunction{T}, MOI.RotatedSecondOrderCone}
+    dimension::Int  # dimension of the SOC constraint
+    less_than::Bool # whether the constraint was ≤ or ≥
+    set_constant::T # the constant that was on the set
+end
+function QuadtoSOCBridge{T}(model, func::MOI.ScalarQuadraticFunction{T},
+                            set::Union{MOI.LessThan{T},
+                                       MOI.GreaterThan{T}}) where T
+    set_constant = MOIU.getconstant(set)
+    less_than = set isa MOI.LessThan
+    if !less_than
+        set_constant = -set_constant
+    end
+    Q, index_to_variable_map = matrix_from_quadratic_terms(func.quadratic_terms)
+    @show Q
+    if !less_than
+        rmul!(Q, -1)
+    end
+    @show Q
+    U = cholesky(Symmetric(Q)).U
```

Comment:
I'm concerned about the use of a dense factorization. This is probably the wrong thing to do when `Q` isn't trivially small. (< 100x100). The fact that this bridge will be applied automatically means that users are going to run into a surprise performance bomb with no feedback about what was wrong.